### PR TITLE
fix: Make NEXT_PUBLIC_BUTTER_CMS_API_KEY optional

### PR DESCRIPTION
**Pull request by soniaklimas**
I want to merge this change because it makes NEXT_PUBLIC_BUTTER_CMS_API_KEY env optional

# Pull Request Checklist

- [ ] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

**Review by karolkarolka — APPROVED** [review]

File: packages/infrastructure/src/public/butter-cms/cms-menu/types.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
 export type ButterCMSMenuServiceConfig = {
-  token: string;
+  token?: string;
```
Comment (karolkarolka) [inline packages/infrastructure/src/public/butter-cms/cms-menu/types.ts:2]: token is required in infra but the infra shouldn't run without that env variable which can be undefined. It isn't necessary to run the storefront

**Review by karolkarolka — CHANGES_REQUESTED** [review]

File: packages/infrastructure/src/public/butter-cms/cms-menu/types.ts (hunk 1 above)
Comment (soniaklimas) [reply packages/infrastructure/src/public/butter-cms/cms-menu/types.ts:2]: I updated cms service to throw an error if no token.

File: apps/storefront/src/services/cms.ts [hunk 2]
```diff
@@ -15,6 +15,12 @@ const getCMSPageService = async (): Promise<CMSPageService> => {
       apiURL: clientEnvs.NEXT_PUBLIC_SALEOR_API_URL,
     });
   } else {
+    if (!clientEnvs.NEXT_PUBLIC_BUTTER_CMS_API_KEY) {
```
Comment (karolkarolka) [inline apps/storefront/src/services/cms.ts:18]: use `import { invariant } from "ts-invariant";` for that, lookup for usage within the project

File: packages/infrastructure/src/public/butter-cms/cms-page/types.ts [hunk 3]
```diff
@@ -1,3 +1,3 @@
 export type ButterCMSPageServiceConfig = {
-  token: string;
+  token?: string;
```
Comment (karolkarolka) [inline packages/infrastructure/src/public/butter-cms/cms-page/types.ts:2]:
```suggestion
  token: string;
```

File: packages/infrastructure/src/public/butter-cms/cms-menu/types.ts (hunk 1 above)
Comment (karolkarolka) [inline packages/infrastructure/src/public/butter-cms/cms-menu/types.ts:2]:
```suggestion
  token: string;
```

**Review by karolkarolka — CHANGES_REQUESTED** [review]

**Review by karolkarolka — APPROVED** [review]

**Pull request merged**